include/nlohmann/detail/json_pointer.hpp

```diff
@@ -114,6 +113,16 @@ class json_pointer
         return last;
     }
 
+    /*!
+    @brief remove and return last reference pointer
+    @throw out_of_range.405 if JSON pointer has no parent
+    */
+    void push_back(const std::string& tok)
+    {
+        reference_tokens.push_back(tok);
```

Comment:
Don't you need to make sure that the token is valid wrt. to the JSON Pointer grammar. I think tildes need to be escaped somehow. You would just pass the given string to the tokens without check.

Reply:
No, escaping is not needed when pushing back a key-value here, because escaping is done in the to_string()-function. It is not the single key which is escaped in a json-pointer, but the json-pointer it its slashed representation.

Here we are adding an element which is a non-escaped key. As we get a non-escaped when calling pop_back.

I don't know if there are any constraints regarding JSON-keys except than being strings. If there are, this would need to be checked here.